pyfar/dsp/dsp.py

```diff
@@ -187,3 +187,126 @@ def spectrogram(signal, dB=True, log_prefix=20, log_reference=1,
     times -= times[0]
 
     return frequencies, times, spectrogram
+
+
+def regularized_spectrum_inversion(
+        signal, freq_range,
+        regu_outside=1., regu_inside=10**(-200/20), regu_final=None):
+    r"""Invert the spectrum of a signal applying frequency dependent
+    regularization. Regularization can either be specified within a given
+    frequency range using two different regularization factors, or for each
+    frequency individually using the parameter `regu_final`. In the first case
+    the regularization factors for the frequency regions are cross-faded using
+    a raised cosine window function with a width of `math:f*\sqrt(2)` above and
+    below the given frequency range. Note that the resulting regularization
+    function is adjusted to the quadratic maximum of the given signal.
+    In case the `regu_final` parameter is used, all remaining options are
+    ignored and an array matching the number of frequency bins of the signal
+    needs to be given. In this case, no normalization of the regularization
+    function is applied. Finally, the inverse spectrum is calculated as
+    [1]_, [2]_,
+
+    .. math::
+
+        S^{-1}(f) = \frac{S^*(f)}{S^*(f)S(f) + \epsilon(f)}
+
+
+    Parameters
+    ----------
+    signal : pyfar.Signal
+        The signals which spectra are to be inverted.
+    freq_range : tuple, array_like, double
+        The upper and lower frequency limits outside of which the
+        regularization factor is to be applied.
+    regu_outside : float, optional
+        The normalized regularization factor outside the frequency range.
+        The default is 1.
+    regu_inside : float, optional
+        The normalized regularization factor inside the frequency range.
+        The default is 10**(-200/20).
+    regu_final : float, array_like, optional
+        The final regularization factor for each frequency, by default None.
+        If this parameter is set, the remaining regularization factors are
+        ignored.
+
+    Returns
+    -------
+    pyfar.Signal
+        The resulting signal after inversion.
+
+    References
+    ----------
+    .. [1]  O. Kirkeby and P. A. Nelson, “Digital Filter Designfor Inversion
+            Problems in Sound Reproduction,” J. Audio Eng. Soc., vol. 47,
+            no. 7, p. 13, 1999.
+
+    .. [2]  P. C. Hansen, Rank-deficient and discrete ill-posed problems:
+            numerical aspects of linear inversion. Philadelphia: SIAM, 1998.
+
+    """
+    data = signal.freq
```

Comment:
How about two quick checks:
- `isintance(signal, Signal)`
- `freq_range.size == 1` (in the if case)

Reply:
:+1: